### ISSUE TITLE
FormatException: Invalid radix-8 number

### ISSUE THREAD

**Issue by isoos** [brendan-duncan/archive#34]
I'm validating your library to check whether we could using it in pub, instead of invoking the external `tar` command every time. I've found the `FormatException: Invalid radix-8 number` exception with the following stacktrace:

````
dart:core                                     int.parse
package:archive/src/tar/tar_file.dart 177:17  TarFile._parseInt
package:archive/src/tar/tar_file.dart 65:15   TarFile.TarFile.read
package:archive/src/tar_decoder.dart 23:24    TarDecoder.decodeBuffer
package:archive/src/tar_decoder.dart 10:12    TarDecoder.decodeBytes
````

I've also created a simple test for it:

````
import 'dart:io';

import 'package:archive/archive.dart';
import 'package:test/test.dart';

void main() {
  group('pub archives', () {
    HttpClient client;
    setUpAll(() {
      client = new HttpClient();
    });
    tearDownAll(() {
      client.close(force: true);
    });

    test('logfmt 0.4.0', () async {
      final HttpClientRequest rq = await client.getUrl(Uri.parse(
          'https://storage.googleapis.com/pub-packages/packages/logfmt-0.4.0.tar.gz'));
      final HttpClientResponse rs = await rq.close();
      final List<int> data = (await rs.toList())
          .fold([], (List<int> a, List<int> b) => a..addAll(b));
      expect(data.length, 10240);

      final Archive archive =
          new TarDecoder().decodeBytes(new GZipDecoder().decodeBytes(data));
      expect(archive.toList(), isNotEmpty);
    });
  });
}
````

Please let me know what you think of this issue. I can provide further .tar.gz examples, which have the same parsing error message.

**Comment by brendan-duncan** [comment]
Looks like I need to add support for the tar PAX extension. I at least pushed a fix the crash, but until I add the PAX extension support, the PAX headers will get treated as files so you'll have extra files that get extracted out (in directories called PaxHeader). These are similar to the meta files Mac's use. I'll add the proper PAX extension support shortly.

**Comment by brendan-duncan** [comment]
I pushed a fix to handle the PAX header files that are part of the POSIX extension. I still need to handle parsing them to extract the extra metadata, but the only thing it really means is the group and user id for files may not be correct, or some other more extreme cases that would not occur in pub archives, like files exceeding 8GB or extremely long file names. I will continue working on trying to conform to the full spec, but I suspect as it is now is sufficient for the archives in pub.

**Comment by isoos** [comment]
[tarurls.txt](https://github.com/brendan-duncan/archive/files/1028333/tarurls.txt)
I've uploaded a few URLs of .tar.gz files that I had some issues with. I think half of them were parse issues, the other half was parsing OK but the README contained a weird UTF-8. So best case you should be able to parse all of them, worst case roughly half of them.

**Comment by brendan-duncan** [comment]
Thanks. I will ensure they all extract correctly.

Strange that I've had this package available for so long with no issue, and
suddenly all these issues. I think the newer posix standard tar programs
only recently started being used, particularly on Linux.

**Comment by brendan-duncan** [comment]
I noticed 7zip doesn't handle the paxHeaders either and is writing them out as files. The POSIX standard does say this is the proper thing to do if the PAX metadata files aren't explicitly handled, but this seems wrong to me as it adds extra files that wouldn't otherwise exist. I will continue to have the archive library not write those paxHeader files out, and since I'm using 7zip to validate the extracted archives, I'll have my tests ignore those paxHeader files from the 7zip extractions.

**Comment by brendan-duncan** [comment]
I ran a file-by-file, byte-by-byte comparison test for every archive in the tarurls.txt list, extracting them with 7zip and the dart library. Every archive passed the test.  The test is in test/pub_test.dart.  

The pushed version has the 7zip and file comparison functions disabled since the Travis CI system won't have 7zip available. I can change it to use the Linux tar program for doing the CI unit test, if necessary.

**Comment by isoos** [comment]
Thanks @brendan-duncan! If you feel confident about it, please publish the new version, I'll run it through the staging db of pub.

**Comment by brendan-duncan** [comment]
I'll publish the new version as soon as the fix for pub gets released (or
I'll switch it to the non dev version).

**Comment by brendan-duncan** [comment]
I'll publish the new version as soon as the fix for pub gets released (or I'll switch it to the non dev version).

**Comment by brendan-duncan** [comment]
All right, I switched to the stable branch and published the fixes as 1.0.29.

**Comment by isoos** [comment]
I've validated the pub staging, and this issue is fixed with 1.0.29.
